pkg/kubectl/plugins/loader.go

```diff
@@ -88,8 +88,15 @@ func (l *DirectoryPluginLoader) Load() (Plugins, error) {
 			return nil
 		}
 
-		plugin.Dir = filepath.Dir(path)
-		plugin.DescriptorName = fileInfo.Name()
+		var setSource func(path string, fileInfo os.FileInfo, p *Plugin)
+		setSource = func(path string, fileInfo os.FileInfo, p *Plugin) {
+			p.Dir = filepath.Dir(path)
+			p.DescriptorName = fileInfo.Name()
+			for _, child := range p.Tree {
+				setSource(path, fileInfo, child)
+			}
+		}
+		setSource(path, fileInfo, plugin)
```

Comment:
I have them in 2 separate commits here, but I'll remember that for the next ones. ;)